sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1328,6 +1332,155 @@ void captureSessionToken(RxDocumentServiceRequest request, RxDocumentServiceResp
+
+                             //Create the range query map that contains the query to be run for that partitionkeyrange
+                             Map<PartitionKeyRange, String> rangeQueryMap;
+                             rangeQueryMap = getRangeQueryMap(partitionRangeItemKeyMap, collection.getPartitionKey());
+
+                             String sqlQuery = "this is dummy and only used for creating " +
+                                                   "ParallelDocumentQueryExecutioncontext, but not used";
+
+                             return createReadManyQuery(collectionLink, new SqlQuerySpec(sqlQuery), new FeedOptions(), Document.class,
+                                                        ResourceType.Document, collection, rangeQueryMap)
+                                        .collectList() // aggregating the result construct a FeedResponse and aggregate RUs.
+                                        .map(feedList -> {
+                                            List<Document> finalList = new ArrayList<>();
+                                            HashMap<String, String> headers = new HashMap<>();
+                                            double requestCharge = 0;
+                                            for (FeedResponse<Document> page : feedList) {
+                                                requestCharge += page.getRequestCharge();
+                                                finalList.addAll(page.getResults());
+                                            }
+                                            headers.put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double.toString(requestCharge));
+                                            FeedResponse<Document> frp = BridgeInternal.createFeedResponse(finalList, headers);
+                                            return frp;
+                                        });
+                         });
+                     }
+            );
+      
+    }
+
+    // TODO: This needs refactoring and optimization and betterment and everything
+    private Map<PartitionKeyRange, String> getRangeQueryMap(
+        Map<PartitionKeyRange, List<Pair<String, PartitionKey>>> partitionRangeItemKeyMap,
+        PartitionKeyDefinition partitionKeyDefinition) {
+        //TODO: Optimise this
+        String partitionKeyPathString = partitionKeyDefinition.getPaths().get(0);
+        String partitionKeyPath = (new StringBuilder(partitionKeyPathString)).deleteCharAt(0).toString();
+        
+        Map<PartitionKeyRange, String> rangeQueryMap = new HashMap<>();
+        
+        for (PartitionKeyRange range : partitionRangeItemKeyMap.keySet()) {
+            StringBuilder queryStringBuilder = new StringBuilder();
+            queryStringBuilder.append("select * from c where ");
+            List<Pair<String, PartitionKey>> pairsList = partitionRangeItemKeyMap.get(range);
+            for (Pair p : pairsList) {
+                queryStringBuilder.append("(c.id = \"");
```

Comment:
we can create the template of query using StringBuilder, but for the values of id and pk, can we use SqlParameter?

that way you don't have to worry about serializing the query and also it is safe against sql query injection attack?

Reply:
Thank you for the suggestion, moved to use  SqlQuerySpec and SqlParameters, looks much cleaner and also helps handle some additional types without heave conversions